server/services/functionalMailboxService.ts

```diff
@@ -0,0 +1,109 @@
+import * as R from 'ramda'
+import { LduMap, LduWithTeams, LocalDeliveryUnitDto, ProbationTeamsClient } from '../../types/probationTeams'
+import { DeliusClient, Ldu, ProbationArea, ProbationTeam } from '../../types/delius'
+
+export const mergeLduData = (
+  ldus: Ldu[],
+  lduDtos: {
+    [localDeliveryUnitCode: string]: LocalDeliveryUnitDto
+  }
+): LduMap => {
+  const lduMap = R.pipe(
+    R.indexBy(R.prop('code')),
+    R.map((ldu) => ({ description: ldu.description }))
+  )(ldus)
+
+  const filteredDtos = R.map(R.pick(['functionalMailbox']), lduDtos)
+
+  return R.mergeDeepRight(lduMap, filteredDtos)
+}
+
+function getDescription(ldus: any[], lduCode: string) {
+  return R.pathOr(
+    '',
+    [0, 'description'],
+    ldus.filter((ldu) => ldu.code === lduCode)
+  )
+}
+
+function mergeProbationTeams(probationTeams: ProbationTeam[], localDeliveryUnitDto: LocalDeliveryUnitDto) {
+  return R.pipe(
+    R.indexBy(R.prop('code')),
```

Comment:
Holy Ramda! 

I've been trying to read this method for a few minutes and still don't really understand what it's doing. 

When playing in a ramda it seem like the ` R.indexBy(R.prop('code')),` is unnecessary as straight away you throw it away be doing the map to extract out the description?